lib/libc/Kconfig

```diff
@@ -15,10 +15,12 @@ config SUPPORT_MINIMAL_LIBC
 	bool
 	default y
 
+# Picolibc with C++ support in Zephyr SDK is handled by Zephyr SDK's own Kconfig.
 config PICOLIBC_SUPPORTED
 	bool
 	depends on ARC || ARM || ARM64 || MIPS || RISCV
 	depends on "$(ZEPHYR_TOOLCHAIN_VARIANT)" != "arcmwdt"
+	depends on !(CPP && "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr")
```

Comment:
Sorry for the late review.

I think we have tried to work backwards on this and ended up with something quite convoluted -- what we really should be specifying here is _when **GLIBCXX** (aka. libstdc++) is supported_, not _when **PICOLIBC** is supported._

GLIBCXX is supported when the toolchain provides the GLIBCXX specifically built for the selected LIBC. This means:

* When NEWLIB (toolchain-provided) is selected, GLIBCXX is supported if the toolchain provides a GLIBCXX built for NEWLIB.
* When PICOLIBC (toolchain-provided) is selected, GLIBCXX is supported if the toolchain provides a GLIBCXX built for PICOLIBC.

Since NEWLIB, when available, is currently only toolchain-provided and a C/C++ toolchain that includes NEWLIB almost certainly includes the GLIBCXX built for NEWLIB, we simply specify `GLIBCXX_LIBCPP depends on NEWLIB_LIBC`; but, in reality, it should really be something like `GLIBCXX_LIBCPP depends on NEWLIB_LIBC && TOOLCHAIN_HAS_GLIBCXX_NEWLIB`.

Similarly, for PICOLIBC, the condition should essentially be `GLIBCXX_LIBCPP depends on PICOLIBC && TOOLCHAIN_HAS_GLIBCXX_PICOLIBC`. But, unlike NEWLIB, PICOLIBC can also be non-toolchain-provided, in which case the GLIBCXX built for the PICOLIBC is unavailable, so it needs to be `GLIBCXX_LIBCPP depends on PICOLIBC && !PICOLIBC_USE_MODULE && TOOLCHAIN_HAS_GLIBCXX_PICOLIBC`.

So, ideally, we should have ended up with something like:

```kconfig
config GLIBCXX_LIBCPP
	bool "GNU C++ Standard Library"
	depends on !NATIVE_APPLICATION
	depends on (NEWLIB_LIBC && TOOLCHAIN_HAS_GLIBCXX_NEWLIB) ||
		   (PICOLIBC && !PICOLIBC_USE_MODULE && TOOLCHAIN_HAS_GLIBCXX_PICOLIBC)
```

where `TOOLCHAIN_HAS_GLIBCXX_NEWLIB` and `TOOLCHAIN_HAS_GLIBCXX_PICOLIBC` are set by the Zephyr SDK CMake package (or automatic toolchain feature detection logic in the future).

Also, C++ library, in its current form, can be considered an extension to the C library and should exist as an upper layer of the C library support; so, the libc choice should dictate which libcpp can be selected, not the other way around.

Since we are near the feature freeze for 3.3, I will leave this alone. I will rework this in the future to make it more logically consistent and straight forward.